### PR TITLE
Add build support for arm64

### PR DESCRIPTION
**Pull request by Firefishy**
THIS IS FOR COMMENT. Needs some clean-up / refactor work.

## What I am changing
- ARM64 build of the gunicorn container
- Update GitHub Actions to latest releases
- Simplify the container tagging by using `docker/metadata-action` (also fix a case sensitive bug on fork)
- Add if statements to allow actions to cleanly run if forked
- Add dependabot to keep GHA updated

## How I did it
- TBC

## How you can test it
- https://github.com/Firefishy/titiler/actions/runs/7186402001/job/19571794840

## Related Issues
- https://github.com/developmentseed/titiler/discussions/387

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: github-actions
+    directory: /
+    schedule:
+      interval: daily
```
Comment (vincentsarago) [inline .github/dependabot.yml:6]: can we do less than daily? I don't think the actions versions are super important that they deserve a daily check/update

**Comment by vincentsarago** [comment]
overall looks good 🙏

**Comment by Firefishy** [comment]
> overall looks good 🙏

Cool. I'll refactor and clean it up.

GitHub actions check schedule I'll set to weekly. True update frequency is driven by when upstream have new major releases.

**Comment by vincentsarago** [comment]
@Firefishy can we merge this?

File: docker-compose.yml [hunk 2]
```diff
@@ -48,7 +48,7 @@ services:
 
   titiler-uvicorn:
     # See https://github.com/developmentseed/titiler/discussions/387
-    platform: linux/amd64
+    # platform: linux/amd64
```
Comment (vincentsarago) [inline docker-compose.yml:51]:
```suggestion
```

**Comment by vincentsarago** [comment]
rasterio still doesn't provide wheels for Mac OS AMD64 so this PR needs to wait for https://github.com/rasterio/rasterio-wheels/issues/69

**Comment by Firefishy** [comment]
@vincentsarago Sorry, I have been away in South Africa for a month. I am now back and can get this PR into shape.

**Comment by vincentsarago** [comment]
I've commented out `linux/arm64` because we still need rasterio to provide the wheels.

I'm going to merge this PR because I think it provides nice enhancements to the CI

**Comment by jmlord** [comment]
> I've commented out `linux/arm64` because we still need rasterio to provide the wheels.
> 
> I'm going to merge this PR because I think it provides nice enhancements to the CI

In case someone lands here like I did, `linux/arm64`  was later enabled in https://github.com/developmentseed/titiler/commit/a5a48c0994ef1f7ce3e38ab6e7a1a15f47620349 of PR https://github.com/developmentseed/titiler/pull/1286

**Pull request merged**